felt/src/arbitrary.rs

```diff
@@ -0,0 +1,39 @@
+use num_bigint::BigUint;
+use num_traits::Zero;
+use proptest::prelude::*;
+
+use crate::{
+    bigint_felt::{FeltBigInt, FIELD_HIGH, FIELD_LOW},
+    Felt252,
+};
+
+pub(crate) fn any_felt_big_int<const PH: u128, const PL: u128>(
+) -> impl Strategy<Value = FeltBigInt<PH, PL>> {
+    (0..=PH)
+        // turn range into `impl Strategy`
+        .prop_map(|x| x)
+        // choose second 128-bit limb capped by first one
+        .prop_flat_map(|high| {
+            let low = if high == PH {
+                Strategy::sboxed((0..PH).prop_map(|x| x))
+            } else {
+                Strategy::sboxed(any::<u128>())
+            };
+            (Just(high), low)
+        })
```

Comment:
I'm not sure I follow this logic. Wouldn't it make more sense to enable `BigUint`'s implementation of `Arbitrary` and giving it a bound of `(PH << 128) | PL`?

Reply:
The thing is, we need `proptest::Arbitrary`, and `BigUint` only implements _quickcheck_'s and _arbitrary_'s.

Reply:
I thought it used [the arbitrary crate's version](https://github.com/rust-fuzz/arbitrary), which `num_bigint` [also implements](https://github.com/rust-num/num-bigint/blob/master/src/biguint/arbitrary.rs#L21-L34). But looking at the source code it seems it uses its own as you said.
Re: the logic. I think the reason I don't get it is that inside the `if` there's a typo. It should be `(0..PL)` rather than `(0..PH)`.

Reply:
Fixed